packages/clay-button/src/ClayButton.tsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+
+export const DISPLAY_TYPES = ['primary', 'secondary', 'link', 'unstyled'];
+
+interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+	block?: boolean;
+	displayType?: typeof DISPLAY_TYPES[number];
+	id?: string;
+	monospaced?: boolean;
+	small?: boolean;
+	title?: string;
+}
+
+interface IClayButton extends React.FunctionComponent<Props> {
+	Group: typeof ClayButtonGroup;
+}
+
+const ClayButton: IClayButton = ({
+	block,
+	children,
+	className,
+	displayType = 'primary',
+	monospaced,
+	small,
+	type = 'button',
+	...otherProps
+}) => (
+	<button
+		{...otherProps}
+		className={classNames(className, 'btn', `btn-${displayType}`, {
+			'btn-block': block,
+			'btn-monospaced': monospaced,
+			'btn-sm': small,
+		})}
+		type={type}
+	>
+		{children}
```

Comment:
I would probably expect the Button to provide itself the option of rendering an icon on the left or right side without having to pass an icon component.

We can provide the two if necessary, a `ButtonLite`(ButtonBase...) and `Button` with the same behaviors expected by the Lexicon.

Reply:
So one of the reasons I decided not to add `Icon` in this component is because there was no specific css classnames that required `Button` to have knowledge of `Icon`. I also wanted to keep the dependency footprint as small as possible and this also requires the usage to be as explicit as possible.

You can get the same outcome of using an Icon by passing it as a child to Button.
```jsx
<Button>
      <Icon>
      {'Click Me!'}
</Button>
```

Reply:
Yes I really like to use it that way, we just have to be careful not to break the expected behaviors so much and let Dev have to do a lot to achieve the behavior.

We should just remember to add these changes in the "Migrate the Clay components from v2 to v3" tab.

Reply:
Yeah thats actually a good reminder that we should probably document the changes from v2 somewhere with each new component. Should this just be added to `clayui.com`?

Reply:
I'm thinking of creating a guide for migrating from v2 to v3, it will be available in the new clayui.com document in #1525 and maybe a redirect link from version v2 to v3.

Reply:
That sounds good to me. For the time being, where should we put these bits of into? Start a google doc?

Reply:
Yes I think we should start to add this information somewhere as we create new components. I think we can create the `migrate-the-clay-components-from-v2-to-v3.md` file and add it to the root folder maybe, then we can move to clayui.com, I think we'll have more control if this file is here.

Reply:
👍 I went ahead and added that file, we can always remove it later.